### ISSUE TITLE
segment fault on battle_train.py

### ISSUE THREAD

**Issue by marvin-hyk** [leehoon7/IE579_Final_Project#1]
I ran your code 'battle_train.py' twice. all times I've got the segment fault. and Random win stick to 0.000

-----------------------
.....
RL win: 0.768 | Random win: 0.000
RL win: 0.769 | Random win: 0.000
RL win: 0.769 | Random win: 0.000
RL win: 0.770 | Random win: 0.000
RL win: 0.770 | Random win: 0.000
RL win: 0.771 | Random win: 0.000
RL win: 0.772 | Random win: 0.000
RL win: 0.772 | Random win: 0.000
RL win: 0.773 | Random win: 0.000
[2]    102079 segmentation fault (core dumped)  python battle_train.py

**Comment by leehoon7** [comment]
It makes sense for 'Random' to remain at 0 because it's quite challenging to eliminate all enemies within the time limit defined by the random policy. When I run it multiple times, there were instances where the random policy win the game in the early stages of learning.

I reran it about three times on my end, and in my case, no segmentation faults occurred. It is possible that error is caused by a differneces in settings. So, I would appreciate it if you could provide information on the environment settings (Linux and Python version) In my case, I am using Ubuntu 18.04.6 LTS and Python 3.7.16.

Could you please check for any additional error logs? Specifically, are there any additional logs relevant to a segmentation fault? Thank you.

Best regards,
Kanghoon.

**Comment by marvin-hyk** [comment]
My environment is Ubuntu 22.04.1, Python 3.7.12 of Miniconda. (I just followed your pdf instruction.)
It occasionally comes from python package version conflict. So, I left my package list as below.
Would you provide me your package list as 'Requirements.txt' for easy environment setup? 
I'll let you know that your env is working on my machine.

Hee-young

-----------------------------------------------
packages in environment at /home/************/miniconda3/envs/game_theory2:

Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       2_gnu    conda-forge
absl-py                   2.0.0                    pypi_0    pypi
astor                     0.8.1                    pypi_0    pypi
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                pyhd8ed1ab_3    conda-forge
backports.functools_lru_cache 1.6.5              pyhd8ed1ab_0    conda-forge
blas                      2.16                        mkl    conda-forge
brotli-python             1.0.9            py37hd23a5d3_7    conda-forge
bzip2                     1.0.8                hd590300_5    conda-forge
ca-certificates           2023.08.22           h06a4308_0  
certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
charset-normalizer        3.3.2              pyhd8ed1ab_0    conda-forge
conda-pack                0.6.0              pyhd3eb1b0_0  
cudatoolkit               11.6.0              hecad31d_10    conda-forge
cycler                    0.11.0                   pypi_0    pypi
debugpy                   1.6.0            py37hd23a5d3_0    conda-forge
decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
entrypoints               0.4                pyhd8ed1ab_0    conda-forge
ffmpeg                    4.3                  hf484d3e_0    pytorch
fonttools                 4.38.0                   pypi_0    pypi
freetype                  2.12.1               h267a509_2    conda-forge
gast                      0.5.4                    pypi_0    pypi
gmp                       6.2.1                h58526e2_0    conda-forge
gnutls                    3.6.13               h85f3911_1    conda-forge
grpcio                    1.59.2                   pypi_0    pypi
h5py                      3.8.0                    pypi_0    pypi
idna                      3.4                pyhd8ed1ab_0    conda-forge
imageio                   2.31.2                   pypi_0    pypi
imageio-ffmpeg            0.4.9                    pypi_0    pypi
importlib-metadata        6.7.0                    pypi_0    pypi
intel-openmp              2023.1.0         hdb19cb5_46306  
ipykernel                 6.16.2             pyh210e3f2_0    conda-forge
ipython                   7.33.0           py37h89c1867_0    conda-forge
jedi                      0.19.1             pyhd8ed1ab_0    conda-forge
jpeg                      9e                   h0b41bf4_3    conda-forge
jupyter_client            7.4.9              pyhd8ed1ab_0    conda-forge
jupyter_core              4.11.1           py37h89c1867_0    conda-forge
keras-applications        1.0.8                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.4.5                    pypi_0    pypi
lame                      3.100             h166bdaf_1003    conda-forge
lcms2                     2.14                 h6ed2654_0    conda-forge
ld_impl_linux-64          2.38                 h1181459_1  
lerc                      3.0                  h9c3ff4c_0    conda-forge
libblas                   3.8.0                    16_mkl    conda-forge
libcblas                  3.8.0                    16_mkl    conda-forge
libdeflate                1.8                  h7f98852_0    conda-forge
libffi                    3.4.4                h6a678d5_0  
libgcc-ng                 13.2.0               h807b86a_3    conda-forge
libgfortran-ng            7.5.0               h14aa051_20    conda-forge
libgfortran4              7.5.0               h14aa051_20    conda-forge
libgomp                   13.2.0               h807b86a_3    conda-forge
libiconv                  1.17                 h166bdaf_0    conda-forge
liblapack                 3.8.0                    16_mkl    conda-forge
liblapacke                3.8.0                    16_mkl    conda-forge
libnsl                    2.0.1                hd590300_0    conda-forge
libpng                    1.6.39               h753d276_0    conda-forge
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libstdcxx-ng              11.2.0               h1234567_1  
libtiff                   4.4.0                hecacb30_2  
libwebp-base              1.3.2                hd590300_0    conda-forge
libxcb                    1.13              h7f98852_1004    conda-forge
libzlib                   1.2.13               hd590300_5    conda-forge
lz4-c                     1.9.3                h9c3ff4c_1    conda-forge
markdown                  3.4.4                    pypi_0    pypi
markupsafe                2.1.3                    pypi_0    pypi
matplotlib                3.5.3                    pypi_0    pypi
matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
mkl                       2020.2                      256  
mock                      5.1.0                    pypi_0    pypi
ncurses                   6.4                  h6a678d5_0  
nest-asyncio              1.5.8              pyhd8ed1ab_0    conda-forge
nettle                    3.6                  he412f7d_0    conda-forge
numpy                     1.21.6           py37h976b520_0    conda-forge
openh264                  2.1.1                h780b84a_0    conda-forge
openjpeg                  2.5.0                h7d73246_0    conda-forge
openssl                   3.1.4                hd590300_0    conda-forge
packaging                 23.2               pyhd8ed1ab_0    conda-forge
parso                     0.8.3              pyhd8ed1ab_0    conda-forge
pexpect                   4.8.0              pyh1a96a4e_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    9.5.0                    pypi_0    pypi
pip                       22.3.1           py37h06a4308_0  
prompt-toolkit            3.0.40             pyha770c72_0    conda-forge
protobuf                  3.20.1                   pypi_0    pypi
psutil                    5.9.3            py37h540881e_0    conda-forge
pthread-stubs             0.4               h36c2ea0_1001    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pygame                    2.5.2                    pypi_0    pypi
pygments                  2.16.1             pyhd8ed1ab_0    conda-forge
pyparsing                 3.1.1                    pypi_0    pypi
pysocks                   1.7.1            py37h89c1867_5    conda-forge
python                    3.7.12          hf930737_100_cpython    conda-forge
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python_abi                3.7                     4_cp37m    conda-forge
pytorch                   1.12.0          py3.7_cuda11.6_cudnn8.3.2_0    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pyzmq                     23.0.0           py37h0c0c2a8_0    conda-forge
readline                  8.2                  h5eee18b_0  
requests                  2.31.0             pyhd8ed1ab_0    conda-forge
setuptools                65.6.3           py37h06a4308_0  
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.41.2               h5eee18b_0  
tensorboard               1.13.1                   pypi_0    pypi
tensorflow                1.13.1                   pypi_0    pypi
tensorflow-estimator      1.13.0                   pypi_0    pypi
termcolor                 2.3.0                    pypi_0    pypi
tk                        8.6.12               h1ccaba5_0  
torch                     1.13.1+cpu               pypi_0    pypi
torchaudio                0.12.0               py37_cu116    pytorch
torchvision               0.13.0               py37_cu116    pytorch
tornado                   6.2              py37h540881e_0    conda-forge
traitlets                 5.9.0              pyhd8ed1ab_0    conda-forge
typing_extensions         4.7.1              pyha770c72_0    conda-forge
urllib3                   2.0.7              pyhd8ed1ab_0    conda-forge
wcwidth                   0.2.9              pyhd8ed1ab_0    conda-forge
werkzeug                  2.2.3                    pypi_0    pypi
wheel                     0.38.4           py37h06a4308_0  
xorg-libxau               1.0.11               hd590300_0    conda-forge
xorg-libxdmcp             1.1.3                h7f98852_0    conda-forge
xz                        5.4.2                h5eee18b_0  
zeromq                    4.3.4                h9c3ff4c_1    conda-forge
zipp                      3.15.0                   pypi_0    pypi
zlib                      1.2.13               hd590300_5    conda-forge
zstd                      1.5.2                h8a70e8d_1    conda-forge

**Comment by leehoon7** [comment]
Thanks, here's a list of the packages I installed. (I saved it as a `req.yaml`)
maybe you can install it using `conda env create -f req.yaml`.
(BTW, I noticed that your Ubuntu version is different from mine. I'm not sure if that could be causing any issues.)

```
name: game_theory
channels:
  - defaults
dependencies:
  - _libgcc_mutex=0.1=main
  - _openmp_mutex=5.1=1_gnu
  - ca-certificates=2023.08.22=h06a4308_0
  - certifi=2022.12.7=py37h06a4308_0
  - ld_impl_linux-64=2.38=h1181459_1
  - libffi=3.4.4=h6a678d5_0
  - libgcc-ng=11.2.0=h1234567_1
  - libgomp=11.2.0=h1234567_1
  - libstdcxx-ng=11.2.0=h1234567_1
  - ncurses=6.4=h6a678d5_0
  - openssl=1.1.1w=h7f8727e_0
  - pip=22.3.1=py37h06a4308_0
  - python=3.7.16=h7a1cb2a_0
  - readline=8.2=h5eee18b_0
  - setuptools=65.6.3=py37h06a4308_0
  - sqlite=3.41.2=h5eee18b_0
  - tk=8.6.12=h1ccaba5_0
  - wheel=0.38.4=py37h06a4308_0
  - xz=5.4.2=h5eee18b_0
  - zlib=1.2.13=h5eee18b_0
  - pip:
      - absl-py==2.0.0
      - astor==0.8.1
      - cycler==0.11.0
      - fonttools==4.38.0
      - gast==0.5.4
      - grpcio==1.59.2
      - h5py==3.8.0
      - imageio==2.31.2
      - imageio-ffmpeg==0.4.9
      - importlib-metadata==6.7.0
      - keras-applications==1.0.8
      - keras-preprocessing==1.1.2
      - kiwisolver==1.4.5
      - markdown==3.4.4
      - markupsafe==2.1.3
      - matplotlib==3.5.3
      - mock==5.1.0
      - numpy==1.21.6
      - packaging==23.2
      - pillow==9.5.0
      - protobuf==3.20.1
      - psutil==5.9.6
      - pyparsing==3.1.1
      - python-dateutil==2.8.2
      - six==1.16.0
      - tensorboard==1.13.1
      - tensorflow==1.13.1
      - tensorflow-estimator==1.13.0
      - termcolor==2.3.0
      - torch==1.13.1+cpu
      - typing-extensions==4.7.1
      - werkzeug==2.2.3
      - zipp==3.15.0
```

Best regards,
Kanghoon.

**Comment by marvin-hyk** [comment]
Your env is working. :+1:

**Comment by leehoon7** [comment]
Oh, sounds good! 
Did you resolve the issue by installing new environment with `req.yaml` I shared ?

**Comment by marvin-hyk** [comment]
Yes, it is. But there is an error about torch of your req.yaml. So,

1. I delete torch line on your yaml

2. Create env using your  yaml file

3. and install torch cpu 

pip install torch==1.13.1+cpu torchvision==0.14.1+cpu torchaudio==0.13.1 --extra-index-url https://download.pytorch.org/whl/cpu
